example/poisson/poisson_driver.cpp

```diff
@@ -59,13 +60,84 @@ TaskCollection PoissonDriver::MakeTaskCollection(BlockList_t &blocks) {
     auto &md = pmesh->mesh_data.GetOrAdd("base", i);
     auto &mdelta = pmesh->mesh_data.GetOrAdd("delta", i);
 
     TaskList &tl = solver_region[i];
 
+    //--- Demo a few reductions
+    // pass a pointer to the variable being reduced into
+    auto loc_red = tl.AddTask(none, poisson_package::SumMass<MeshData<Real>>, md.get(),
+                              &total_mass.val);
+    // make it a regional dependency so dependent tasks can't execute until all lists do
+    // this
+    solver_region.AddRegionalDependencies(0, i, loc_red);
+
+    auto rank_red = tl.AddTask(
+        none,
+        [](int *max_rank) {
+          *max_rank = std::max(*max_rank, Globals::my_rank);
+          return TaskStatus::complete;
+        },
+        &max_rank.val);
+    solver_region.AddRegionalDependencies(1, i, rank_red);
+
+    // start a non-blocking MPI_Iallreduce
+    auto start_global_reduce =
+        (i == 0 ? tl.AddTask(loc_red, &AllReduce<Real>::StartReduce, &total_mass, MPI_SUM)
+                : none);
+
+    auto start_rank_reduce =
+        (i == 0 ? tl.AddTask(rank_red, &Reduce<int>::StartReduce, &max_rank, 0, MPI_MAX)
+                : none);
+
+    // test the reduction until it completes
+    auto finish_global_reduce =
+        tl.AddTask(start_global_reduce, &AllReduce<Real>::CheckReduce, &total_mass);
+    solver_region.AddRegionalDependencies(2, i, finish_global_reduce);
+
+    auto finish_rank_reduce =
+        tl.AddTask(start_rank_reduce, &Reduce<int>::CheckReduce, &max_rank);
+    solver_region.AddRegionalDependencies(3, i, finish_rank_reduce);
+
+    // notice how we must always pass a pointer to the reduction value
+    // since tasks capture args by value, this would print zero if we just passed in
+    // the val since the tasks that compute the value haven't actually executed yet
+    auto report_mass = (i == 0 && Globals::my_rank == 0
+                            ? tl.AddTask(
+                                  finish_global_reduce,
+                                  [](Real *mass) {
+                                    std::cout << "Total mass = " << *mass << std::endl;
+                                    return TaskStatus::complete;
+                                  },
+                                  &total_mass.val)
+                            : none);
+    auto report_rank = (i == 0 && Globals::my_rank == 0
+                            ? tl.AddTask(
+                                  finish_rank_reduce,
+                                  [](int *max_rank) {
+                                    std::cout << "Max rank = " << *max_rank << std::endl;
+                                    return TaskStatus::complete;
+                                  },
+                                  &max_rank.val)
+                            : none);
+
+    //--- Begining of tasks related to solving the Poisson eq.
+    auto mat_elem =
+        tl.AddTask(none, poisson_package::SetMatrixElements<MeshData<Real>>, md.get());
```

Comment:
Solving the Poisson equation using these sparse matrix elements isn't the only way nor the most straightforward implementation, it's done here to demonstrate these solver building blocks, correct? Can you add the the comment something like "... solving the Poisson eq. using matrix operations to demonstrate stencil solver"

Reply:
done